tools/argdist.py

```diff
@@ -258,7 +258,7 @@ def _generate_streq_function(self, string):
         char needle[] = %s;
         char haystack[sizeof(needle)];
         bpf_probe_read(&haystack, sizeof(haystack), (void *)str);
-        for (int i = 0; i < sizeof(needle); ++i) {
+        for (int i = 0; i < sizeof(needle)-1; ++i) {
```

Comment:
Thanks, there's actually a couple of places in the Python script that I did the same. Will fix and force-push.